### DISCUSSION TITLE
docs: How to populate and use custom fields on a M:N pivot entity

### DISCUSSION THREAD

**Discussion by ryall** [mikro-orm/mikro-orm#3310] [Ideas]
I don't see any way using the current API to be able to set or order by fields defined in a custom pivot entity. For example, in my case, I have:

```ts
@Entity()
export class A {
  @ManyToMany({
    entity: () => B,
    pivotEntity: () => AB,
    inversedBy: (b) => b.a,
    joinColumn: 'a_id',
    inverseJoinColumn: 'b_id',
  })
  b = new Collection<B>(this);
}

@Entity()
export class B extends {
  @ManyToMany({
    entity: () => A,
    pivotEntity: () => AB,
    mappedBy: (a) => a.b,
  })
  a = new Collection<A>(this);
}

@Entity()
export class AB {
  @ManyToOne({
    entity: () => A,
    primary: true,
  })
  a: A;

  @ManyToOne({
    entity: () => B,
    primary: true,
  })
  b: B;

  @Property()
  sortKey: string; // I want to be able to use this without accessing AB directly if possible
}
```

How would I set the `sortKey` on this and how would I then order by that field later, without using the pivot table entity directly?

**Comment by B4nan** [comment]
This is not possible, M:N with pivot entities only offer you to have a M:N collection, to access additional pivot entity properties you need to query that pivot entity directly (or have the 1:m/m:1 properties defined and use those).

I would be happy to consider this feature but I don't see any good way to make this type safe.
